client/gutenberg/extensions/markdown/editor.js

```diff
@@ -15,46 +15,50 @@ import save from './save';
 import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
 import JetpackBlockType from 'gutenberg/extensions/presets/jetpack/utils/jetpack-block-type';
 
-const MarkdownBlock = new JetpackBlockType( 'markdown', {
-	title: __( 'Markdown' ),
-
-	description: (
-		<Fragment>
-			<p>{ __( 'Use regular characters and punctuation to style text, links, and lists.' ) }</p>
-			<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
-				{ __( 'Support reference' ) }
-			</ExternalLink>
-		</Fragment>
-	),
-
-	icon: (
-		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
-			<rect
-				width="198"
-				height="118"
-				x="5"
-				y="5"
-				ry="10"
-				stroke="currentColor"
-				strokeWidth="10"
-				fill="none"
-			/>
-			<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
-		</svg>
-	),
-
-	category: 'jetpack',
-
-	keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
-
-	attributes: {
-		//The Markdown source is saved in the block content comments delimiter
-		source: { type: 'string' },
-	},
+const MarkdownBlock = new JetpackBlockType( {
+	name: 'markdown',
+	requiredBlock: 'markdown',
+	config: {
+		title: __( 'Markdown' ),
+
+		description: (
+			<Fragment>
+				<p>{ __( 'Use regular characters and punctuation to style text, links, and lists.' ) }</p>
+				<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
+					{ __( 'Support reference' ) }
+				</ExternalLink>
+			</Fragment>
+		),
+
+		icon: (
+			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
+				<rect
+					width="198"
+					height="118"
+					x="5"
+					y="5"
+					ry="10"
+					stroke="currentColor"
+					strokeWidth="10"
+					fill="none"
+				/>
+				<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
+			</svg>
+		),
+
+		category: 'jetpack',
 
-	edit,
+		keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
 
-	save,
+		attributes: {
+			//The Markdown source is saved in the block content comments delimiter
+			source: { type: 'string' },
+		},
+
+		edit,
+
+		save,
+	},
 } );
 
-MarkdownBlock.register();
+MarkdownBlock.registerBlock();
```

Comment:
Do we ever not want to register a block? Would wrapping these directly in something like `registerJetpackBlockType()` provide more clear API for developers?